### PR TITLE
Fix a notice with tmpname

### PR DESCRIPTION
**Pull request by luceos**
This fixes a notice with tmpname by creating the tmp dir inside the storage dir

### CODE REVIEW

File: src/Extenders/CreateStorageFolder.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    public function onEnable(Container $container, Extension $extension)
+    {
+        @mkdir(storage_path($this->path));
+    }
+
+    public function onDisable(Container $container, Extension $extension)
+    {
+        // Nee, no, nein, nada, pas de rein.
```
Comment (clarkwinkelmann) [inline src/Extenders/CreateStorageFolder.php]: Is the last part supposed to be French ? Because it has no meaning if it is :thinking:

**Comment by clarkwinkelmann** [comment]
I'm getting

```
Flarum encountered a boot error (Error)
Call to undefined method FoF\Upload\Extenders\CreateStorageFolder::extend()
thrown in /home/clark/Projects/flarum-beta11/vendor/flarum/core/src/Extension/Extension.php on line 122
```

It seems all Flarum extenders have to extend ExtenderInterface. Some of them can implement LifecycleInterface as well, but it's not allowed to only implement LifecycleInterface.

Maybe core should check the extenders implement ExtenderInterface before calling ->extend() on them to show a better error...

**Comment by luceos** [comment]
Yeah, that's right, forgot about that. Should have tested 🤦‍♂

File: src/Extenders/CreateStorageFolder.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    public function extend(Container $container, Extension $extension = null)
+    {
+        // TODO: Implement extend() method.
```
Comment (clarkwinkelmann) [inline src/Extenders/CreateStorageFolder.php]: It's probably best to remove the TODO text I suppose

File: src/Extenders/CreateStorageFolder.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace FoF\Upload\Extenders;
+
+use Flarum\Extend\ExtenderInterface;
+use Flarum\Extend\LifecycleInterface;
+use Flarum\Extension\Extension;
+use Illuminate\Contracts\Container\Container;
+
+class CreateStorageFolder implements LifecycleInterface, ExtenderInterface
```
Comment (clarkwinkelmann) [inline src/Extenders/CreateStorageFolder.php]: In core ExtenderInterface is always put first (alphabetical order) :stuck_out_tongue:

**Comment by clarkwinkelmann** [comment]
:joy: 

I'm still curious about the might-be-French text (first comment at the top of this discussion)

But otherwise looking good :+1:

**Comment by luceos** [comment]
Might be French? You're criticizing my foreign language capabilities? Ouch.

**Comment by clarkwinkelmann** [comment]
If it's French, it translates to "no kidney". Maybe rein is a typo of "rien" which means nothing. But then it would translate to "no part of nothing" :thinking: :thinking: :thinking:

**Comment by luceos** [comment]
Oh I guess that's what I meant.

**Pull request merged**